src/pages/authentication/Login.jsx

```diff
@@ -1,51 +1,116 @@
+      }
+    } catch (error) {
+      alert(
+        "Error Occured: please check credentials or try again later.",
+        error
+      );
+    }
+  };
+
+  const handleLoginClick = (e) => {
+    e.preventDefault();
+
+    if (userData.email === "" && userData.password === "") {
+      alert("please enter appropriate fields");
+    } else {
+      doLoginNetworkCall();
+    }
+  };
+
+  const handleGuestLoginClick = (e) => {
+    e.preventDefault();
+    setUserData({ email: "johndoe@gmail.com", password: "johnDoe123" });
+  };
+
   return (
-      <>
-    <main className="main-container flex ai-start jc-center">
-      <form className="authentication-container flex flex-column ai-left p-md2 m-xs">
-        <h1 className="title m-s m-rl0 fs-3 fw-600">Log in</h1>
-        <div className="input-wrapper m-xxxs m-rl0">
-          <label htmlFor="email">Email</label>
-          <input
-            type="text"
+    <>
+      <main className="main-container flex ai-start jc-center">
+        <form className="authentication-container flex flex-column ai-left p-md2 m-xs">
+          <h1 className="title m-s m-rl0 fs-3 fw-600">Log in</h1>
+
+          <TextInput
+            labelText="Email"
             id="email"
+            name="email"
             placeholder="johndoe@gmail.com"
-            className="input p-xxs m-xxs m-rl0 bd-rad-sm"
+            userData={userData}
+            setUserData={setUserData}
+            type="email"
           />
-          <label className="success-msg m-xxs m-rl0">
-            Success! Email entered is correct.
-          </label>
-        </div>
-        <div className="input-wrapper m-xxxs m-rl0">
-          <label htmlFor="password">Password</label>
-          <input
+
+          <PasswordInput
+            labelText="Password"
             id="password"
-            className="input p-xxs m-xxs m-rl0 bd-rad-sm"
-            type="password"
+            name="password"
             placeholder="password"
+            userData={userData}
+            setUserData={setUserData}
           />
-          <label className="error-msg m-xxs m-rl0">
-            Wrong Password. Try again.
-          </label>
-        </div>
-        <button type="submit" className="btn btn-primary m-xxs m-rl0">
-          LOGIN
-        </button>
-        <p className="m-xxs m-rl0 center-align-text gray-text">
-          New User?{" "}
-          <a className="primary-color-text" href="/pages/signup/signup.html">
-            Sign Up!
-          </a>
-          .
-        </p>
-        <a href="" className="primary-color-text center-align-text m-xxs m-rl0">
-          Use Guest Credentials.
-        </a>
-      </form>
-    </main>
+
+          <button
+            type="submit"
+            className="btn btn-primary m-xxs m-rl0"
+            onClick={handleLoginClick}
+          >
+            LOGIN
+          </button>
+          <p className="m-xxs m-rl0 center-align-text gray-text">
+            New User?{" "}
```

Comment:
The `{" "}` could be removed. It comes because of `prettier` by default at some places.

Reply:
This is intentionally added here to create a space between two inline words